fiona/ogrext.pyx

```diff
@@ -1723,13 +1722,36 @@ cdef class MemoryFileBase:
 
     def _open(self):
         """Ensure that the instance has a valid VSI file handle."""
+        name_b = self.name.encode('utf-8')
         if not self.exists():
-            self._vsif = VSIFOpenL(self.name.encode("utf-8"), "w")
+            self._vsif = VSIFOpenL(name_b, "w")
             VSIFCloseL(self._vsif)
             self._vsif = NULL
 
         if self._vsif == NULL:
-            self._vsif = VSIFOpenL(self.name.encode("utf-8"), self.mode.encode("utf-8"))
+            self._vsif = VSIFOpenL(name_b, self.mode.encode("utf-8"))
+
+    def _ensure_extension(self, drivername=None):
+        """Ensure that the instance's name uses a file extension supported by the driver."""
+        # Avoid a crashing bug with GDAL versions < 2.
+        gdal_version = GDALVersion.runtime()
+        if gdal_version.major < 2:
+            return
```

Comment:
We got a segfault on Travis when this was called. I'm not inclined to try to make Fiona do the right thing with GDAL 1.11.